src/d3d9/d3d9_device.cpp

```diff
@@ -2244,66 +2250,67 @@ namespace dxvk {
     if (unlikely(ShouldRecord()))
       return m_recorder->SetRenderState(State, Value);
 
-    auto& states = m_state.renderStates;
-    DWORD old = states[State];
-
-    bool changed = old != Value;
+    auto& states         = m_state.renderStates;
+    const DWORD oldValue = states[State];
 
-    if (likely(changed)) {
-      const uint32_t vendorId            = m_adapter->GetVendorId();
-      const bool     isNvidia            = vendorId == uint32_t(DxvkGpuVendor::Nvidia);
-      const bool     isAmd               = vendorId == uint32_t(DxvkGpuVendor::Amd);
-      const bool     isIntel             = vendorId == uint32_t(DxvkGpuVendor::Intel);
+    if (likely(Value != oldValue)) {
+      constexpr uint32_t nvidiaVendorId = uint32_t(DxvkGpuVendor::Nvidia);
+      constexpr uint32_t amdVendorId    = uint32_t(DxvkGpuVendor::Amd);
+      constexpr uint32_t intelVendorId  = uint32_t(DxvkGpuVendor::Intel);
 
-      const bool     oldClipPlaneEnabled = IsClipPlaneEnabled();
-      const bool     oldDepthBiasEnabled = IsDepthBiasEnabled();
-      const bool     oldATOC             = !m_isD3D8Compatible ? IsAlphaToCoverageEnabled() : false;
-      const bool     oldNVDB             = !m_isD3D8Compatible ? states[D3DRS_ADAPTIVETESS_X] == uint32_t(D3D9Format::NVDB) : false;
-      const bool     oldAlphaTest        = IsAlphaTestEnabled();
+      const uint32_t vendorId = m_adapter->GetVendorId();
 
       states[State] = Value;
 
       // AMD's driver hack for ATOC, RESZ, INST and CENT (also supported on Nvidia)
-      if (unlikely(State == D3DRS_POINTSIZE &&
-                   !m_isD3D8Compatible && !isIntel)) {
+      if (unlikely(State == D3DRS_POINTSIZE && vendorId != intelVendorId)) {
         // ATOC (AMD specific)
         constexpr uint32_t AlphaToCoverageEnable  = uint32_t(D3D9Format::A2M1);
         constexpr uint32_t AlphaToCoverageDisable = uint32_t(D3D9Format::A2M0);
 
         if ((Value == AlphaToCoverageEnable
-          || Value == AlphaToCoverageDisable) && isAmd) {
-          m_amdATOC = Value == AlphaToCoverageEnable;
-
-          bool newATOC = IsAlphaToCoverageEnabled();
-          bool newAlphaTest = IsAlphaTestEnabled();
+          || Value == AlphaToCoverageDisable) &&
+             vendorId == amdVendorId &&
+             !m_isD3D8Compatible) {
```

Comment:
`m_isD3D8Compatible` is redundant here because `IsAlphaToCoverageEnabled` checks it anyway. So both `atocEnabled` and `m_atocEnabled` are always false.